spec/rubocop/cop/rspec/multiple_describes_spec.rb

```diff
@@ -50,4 +50,139 @@
+          describe '#do_something' do
+          ^^^^^^^^^^^^^^^^^^^^^^^^ Do not use nested describe - try to split them.
+            subject { my_class_instance.foo }
+          end
+        end
+      RUBY
+    end
+
+    it 'check first describe in each nested describe' do
+      expect_offense(<<-RUBY)
+        describe MyClassController, type: :controller do
+          describe 'do_something' do
+          ^^^^^^^^^^^^^^^^^^^^^^^ Do not use nested describe - try to split them.
+            describe 'do_something_in_describe' do; end
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not use nested describe - try to split them.
+          end
+          describe 'do_something_else' do; end
+        end
+      RUBY
+    end
+
+    it 'ignores describe in top level groups' do
+      expect_no_offenses(<<-RUBY)
+        RSpec.describe MyClass, '#do_something' do
+          subject { my_class_instance.foo }
+        end
+        RSpec.describe MyClass, '#do_something_else' do
+          subject { my_class_instance.bar }
+        end
+      RUBY
+    end
+
+    it 'ignores describe when rspec matcher exists' do
+      expect_no_offenses(<<-RUBY)
+        RSpec.describe MyClass do
+          RSpec::Matchers.define :be_a_multiple_of do |expected|
+            match do |actual|
+              actual % expected == 0
+            end
+          end
+
+          describe '#do_something' do
+            subject { my_class_instance.foo }
+          end
+
+          describe '#do_something_else' do
+            subject { my_class_instance.bar }
+          end
+        end
+      RUBY
+    end
+
+    it 'ignores describe when shared example group exists' do
+      expect_no_offenses(<<-RUBY)
+        RSpec.describe MyClass do
+          shared_examples 'behaves' do;end
+
+          describe '#do_something' do
+            subject { my_class_instance.foo }
+          end
+
+          describe '#do_something_else' do
+            subject { my_class_instance.bar }
+          end
+        end
+      RUBY
+    end
+
+    it 'ignores describe when shared example for group exists' do
+      expect_no_offenses(<<-RUBY)
+        RSpec.describe MyClass do
+          shared_examples_for 'behaves' do;end
+
+          describe '#do_something' do
+            subject { my_class_instance.foo }
+          end
+
+          describe '#do_something_else' do
+            subject { my_class_instance.bar }
+          end
+        end
+      RUBY
+    end
+
+    it 'ignores describe when shared context group exists' do
+      expect_no_offenses(<<-RUBY)
+        RSpec.describe MyClass do
+          shared_context 'behaves' do;end
+
+          describe '#do_something' do
+            subject { my_class_instance.foo }
+          end
+
+          describe '#do_something_else' do
+            subject { my_class_instance.bar }
+          end
+        end
+      RUBY
+    end
+
+    it 'ignores describe when shared context group includes describe' do
```

Comment:
Does the content of `shared_examples` make any difference on the cop's behaviour?

Reply:
No, the cop's behavior is not affected by the contents of the shared example.